jujugui/static/gui/src/app/app.js

```diff
@@ -1933,6 +1933,16 @@ YUI.add('juju-gui', function(Y) {
       if (this.get('sandbox')) {
         console.log('switching models is not supported in sandbox');
       }
+      // Optionally update to the new base URL.
+      if (this.get('gisf')) {
+        var state = this.state;
+        var baseUrl = state.get('baseUrl');
+        var unsavedModelUrl = 'sandbox';
+        // Make sure we're not already in an unsaved model.
+        if (baseUrl.indexOf(unsavedModelUrl) < 0) {
```

Comment:
Ok so lets change `unsavedModelUrl` to be something like `gisfDisconnectedModelName` and add code comments for both points so that they are very clear in the future.